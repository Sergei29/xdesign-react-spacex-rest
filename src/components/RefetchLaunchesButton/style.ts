import styled from "styled-components";

export const RefetchLaunchesButtonContainer = styled.div``;
