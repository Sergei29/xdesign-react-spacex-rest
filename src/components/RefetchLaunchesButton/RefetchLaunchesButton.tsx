import React from "react";
// style:
import { RefetchLaunchesButtonContainer } from "./style";

/**
 * @description presentational component
 * @returns {JSX} component markup, refetch lauches list button
 */
const RefetchLaunchesButton = () => {
  return (
    <RefetchLaunchesButtonContainer>Reload data</RefetchLaunchesButtonContainer>
  );
};

export default RefetchLaunchesButton;
