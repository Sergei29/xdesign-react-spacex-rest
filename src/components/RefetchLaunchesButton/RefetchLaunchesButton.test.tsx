import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { Provider } from "react-redux";
import objStore from "../../redux/store/store";
import RefetchLaunchesButton from "./RefetchLaunchesButton";

describe("RefetchLaunchesButton", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={objStore}>
        <RefetchLaunchesButton />
      </Provider>
    );
  });

  it("should render", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
