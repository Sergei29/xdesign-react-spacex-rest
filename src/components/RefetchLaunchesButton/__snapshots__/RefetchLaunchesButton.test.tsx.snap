// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefetchLaunchesButton should match snapshot 1`] = `ShallowWrapper {}`;
